packages/realm/binding/android/build.gradle

```diff
@@ -0,0 +1,97 @@
+import java.nio.file.Paths
+
+buildscript {
+    repositories {
+        google()
+        mavenCentral()
+        mavenLocal()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:7.3.1'
+    }
+}
+apply plugin: 'com.android.library'
+
+/**
+ * Highly simplified implementation of the Node modules resolution algorithm to locate the React Native directory.
+ */
+static def findReactNative(startDir) {
+  def currentPath = startDir.toPath().normalize()
+  while (currentPath) {
+    def candidateFile = currentPath.resolve("node_modules/react-native").toFile()
+    if (candidateFile.exists()) {
+      return candidateFile.path
+    }
+    currentPath = currentPath.getParent()
+  }
+  throw new GradleException("Unable to find the React Native directory")
+}
+
+/**
+ * Replaces all occurances of the file path separator with '/' if it isn't already '/'.
+ */
+def sanitizeFilePath(String filePath) {
+   return File.separatorChar === '/' as char ? filePath : filePath.replace(File.separatorChar, '/' as char)
+}
+
+// We're resolving this only to pass it to CMake for a workaround which relies on a private React Native API / header.
+def REACT_NATIVE_ROOT_DIR = sanitizeFilePath(findReactNative(rootProject.projectDir));
+
+def reactNativeArchitectures() {
+    def value = project.getProperties().get("reactNativeArchitectures")
+    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
+}
+
+
+android {
+    namespace 'io.realm.react'
+    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : 28
+    buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : '28.0.3'
+    buildFeatures {
+        prefab = true
+    }
+    defaultConfig {
+        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 16
+        targetSdkVersion rootProject.hasProperty('targetSdkVersion') ? rootProject.targetSdkVersion : 28
+        externalNativeBuild {
+            cmake {
+                arguments "-DANDROID_STL=c++_shared",
+                          "-DREACT_NATIVE_ROOT_DIR=${REACT_NATIVE_ROOT_DIR}"
+                targets 'realm-js-android-binding'
+                cppFlags '-std=c++20'
+                abiFilters (*reactNativeArchitectures())
+            }
+        }
+    }
+
+    buildTypes {
+        debug {
+            jniDebuggable true
+            packagingOptions {
+                doNotStrip "**/librealm.so"
+            }
+        }
+    }
+
+    externalNativeBuild {
+        cmake {
+            path file('CMakeLists.txt')
+            version '3.22.1'
```

Comment:
We use 3.29.2 in our PR workflow

Reply:
I don't think the version of CMake matters too much here, as long as it meets the minimum requirements of our CMakeList.txt file. We're aligning this with the version with React Native to cause the least amount of friction to developers: https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/build.gradle.kts#L44-L47

Reply:
Why not say 3.22.1 in our PR workflow?

Reply:
In general, I don't see a reason not to. I just wanted to keep the changes of this PR to a minimum and as you pointed out 3.29.2 is what's currently being used: https://github.com/realm/realm-js/blob/b121047ac34e632bac452d1627ab9b8734929018/.github/workflows/pr-realm-js.yml#L116